### PR TITLE
Fixes part of #1367: Remove the boilerplate "isTablet" if statements from the code (Home Part)

### PR DESCRIPTION
**Pull request by MohamedMedhat1998**
## Explanation

- Refactored `HomeFragment` and `TopicListAdapter` code by removing the boilerplate `if` statements that check the type of the device and replacing them with XML values overridden in different configurations
- Fixes part of #1367 
## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt [hunk 1]
```diff
@@ -31,14 +28,7 @@ class TopicListAdapter(
...
 
-  private val orientation = Resources.getSystem().configuration.orientation
-  private val metrics = DisplayMetrics()
-  private var screenWidth = 0
-
-  init {
-    activity.windowManager.defaultDisplay.getMetrics(metrics)
-    screenWidth = metrics.widthPixels
-  }
+  var spanCount = 0
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt]: This should be `private var spanCount = 0` and create a setter function to update this value.

File: app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt [hunk 2]
```diff
@@ -287,55 +203,50 @@ class TopicListAdapter(
...
-        } else {
+        }
+        4 -> {
           when {
-            position % maxItemsInARow == 0 -> marginLayoutParams.setMargins(
+            (position + 1) % spanCount == 0 -> marginLayoutParams.setMargins(
               marginMax,
               marginTopBottom,
-              /* right= */ 0,
+              0,
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt]: `/* right= */ 0`

File: app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt [hunk 3]
```diff
@@ -287,55 +203,50 @@ class TopicListAdapter(
...
+            (position + 1) % spanCount == 2 -> marginLayoutParams.setMargins(
+              marginMin / 2,
+              marginTopBottom,
               marginMin,
               marginTopBottom
             )
-            position % maxItemsInARow == 2 -> marginLayoutParams.setMargins(
-              /* left= */ 0,
+            (position + 1) % spanCount == 3 -> marginLayoutParams.setMargins(
+              0,
```
Comment (rt4914) [inline app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt]: `/* left= */ 0`

**Review by rt4914 — CHANGES_REQUESTED** [review]
Overall the approach is very good. Suggested some nit changes.

File: app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt (hunk 1 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt]: From what I know, Kotlin internally creates a setter and a getter and uses them. If you create a Java class and tried to access the `spanCount`, you will only be able to access it using its getter `getSpanCount()` which had been generated automatically.  
PTAL [here](https://www.programiz.com/kotlin-programming/getters-setters)

Comment (rt4914) [reply app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt]: @MohamedMedhat1998 Yeah that is correct but we should avoid making variables public to other files as much as possible and thats why the separate setter function will make things precise.

Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt]: Ok, I'll fix that

Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt]: Done

File: app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt (hunk 2 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt]: Done

File: app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt (hunk 3 above)
Comment (MohamedMedhat1998) [reply app/src/main/java/org/oppia/app/home/topiclist/TopicListAdapter.kt]: Done

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**